File: C/quest02/ex08/my_string_index.c
#include <stdio.h>
#include <string.h>

int my_string_index(char* param_1, char param_2){
    for(int i=0; i< strlen(param_1); i++){
        if (param_1[i] == param_2) {
            return i;
        }
    }
    return -1;
}

        /*
int main(){
    char str[]="Hello World";
    char c= 'l';
    int result = my_string_index(str, c);
    printf("index of '%s' in '%c' is %d\n", str, c, result);
    return 0;
}       */

//a s t r u m
//      r
//a == r
//97 == 114
//r == r
//114 == 114

File: Ruby/ruby-quest04/ex00/my_string_index.rb
def my_string_index(param_1, param_2)
    # Loop through each character in the param_1 string
    param_1.each_char.with_index do |char, index|
      # Check if the character matches the param_2 character
      if char == param_2
        # Return the index of the character in the param_1 string
        return index
      end
    end
    # If the param_2 character is not found, return -1
    return -1
  end
  